### PR TITLE
Web fix odd alignment of SelectionArea Text when it's a child of a Column

### PR DESCRIPTION
**Pull request by LimaneGaya**
Fixes weird behaviour of SelectionArea when text is insite a Column or TabBarView

*List which issues are fixed by this PR. You must list at least one issue. An issue is not required if the PR fixes something trivial like a typo.*
#148934  #149426 

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hi @LimaneGaya welcome! Thanks for contributing. This will need a test to verify the expected behavior - and to make sure we don't regress it in the future!

**Comment by LimaneGaya** [comment]
Hi @Piinks thank you for reviewing my PR, does this look okay? for the test i was not sure how to do it but that was the only property that changed from working well and not. Please let me know if i can improve anything.

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 1]
```diff
@@ -3594,6 +3594,21 @@ void main() {
     skip: kIsWeb, // [intended] Web uses its native context menu.
   );
 
+
```
Comment (justinmc) [inline packages/flutter/test/widgets/selectable_region_test.dart:3597]: Nit: Too many newlines here.

**Review by justinmc — COMMENTED** [review]
@Renzo-Olivares Do you have any context on this or is this a web thing?

@LimaneGaya You have an analyzer failure, can you take a look at that in the CI checks below?

File: packages/flutter/test/widgets/selectable_region_test.dart (hunk 1 above)
Comment (LimaneGaya) [reply packages/flutter/test/widgets/selectable_region_test.dart]: Thank you @justinmc for reviewing, i deleted those empty spaces.

File: packages/flutter/test/widgets/selectable_region_test.dart [hunk 2]
```diff
@@ -3594,6 +3594,20 @@ void main() {
     skip: kIsWeb, // [intended] Web uses its native context menu.
   );
 
+  testWidgets('Ensure consistent layout of SelectionArea children between platforms', (WidgetTester tester) async {
```
Comment (Renzo-Olivares) [inline packages/flutter/test/widgets/selectable_region_test.dart:3597]: This test passes on the latest `master`, so I think it should probably be more involved. I think making tests similar to the code samples in the related issues is a good place to start. Mainly the test should verify that the previously broken behavior is no longer broken. I think for this issue https://github.com/flutter/flutter/issues/148934 it is sufficient to verify that the `Text` widgets all begin at the same x axis `Offset`.

File: packages/flutter/lib/src/widgets/_platform_selectable_region_context_menu_web.dart [hunk 3]
```diff
@@ -135,6 +135,9 @@ class PlatformSelectableRegionContextMenu extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Stack(
+      // Setting fit to StackFit.passthrough will prevent the child from misaligning on web
+      // and be consist with other platforms.
+      fit: StackFit.passthrough,
       alignment: Alignment.center,
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/widgets/_platform_selectable_region_context_menu_web.dart:141]: I'm not entirely sure about the origins of why we set the alignment to `Alignment.center` here. I tried removing it and didn't notice any glaring issues. cc @chunhtai if you had any context on this.

File: packages/flutter/lib/src/widgets/_platform_selectable_region_context_menu_web.dart [hunk 4]
```diff
@@ -135,6 +135,9 @@ class PlatformSelectableRegionContextMenu extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Stack(
+      // Setting fit to StackFit.passthrough will prevent the child from misaligning on web
+      // and be consist with other platforms.
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/widgets/_platform_selectable_region_context_menu_web.dart:139]: nit: `consist` -> `consistent`.

**Pull request closed without merging**